sklearn/covariance/_shrunk_covariance.py

```diff
@@ -325,31 +340,14 @@ def ledoit_wolf(X, *, assume_centered=False, block_size=1000):
 
     where mu = trace(cov) / n_features
     """
-    X = check_array(X)
-    # for only one feature, the result is the same whatever the shrinkage
-    if len(X.shape) == 2 and X.shape[1] == 1:
-        if not assume_centered:
-            X = X - X.mean()
-        return np.atleast_2d((X**2).mean()), 0.0
-    if X.ndim == 1:
-        X = np.reshape(X, (1, -1))
-        warnings.warn(
-            "Only one sample available. You may want to reshape your data array"
-        )
-        n_features = X.size
-    else:
-        _, n_features = X.shape
+    X = as_float_array(X)
```

Comment:
We don't need this check. We delegate it to the class (done by the call to `_validate_data`.